.github/workflows/daily_common.yml

```diff
@@ -89,6 +89,21 @@ jobs:
         run: |
           nimble install -y --depsOnly
 
+      - name: Compile nimbus using nim-libp2p
+        if: ${{ matrix.platform.os == 'linux' && matrix.cpu == 'amd64' && matrix.nim.ref == 'version-2-2'}}
```

Comment:
not sure if we should put everything into single daily job. we currently have daily tests, but we can have other daily jobs doing other things. similarly how on commit we have different jobs doing different things. 
 
- code organization gets worse: if we start putting everything into single workflow
- dependencies of jobs are not clear: if tests fails, with same specs as compile nimbus, compile nimbus will not happen
- parameters of daily test workflow as specifically designed for nim-libp2p testing
- generic failure notifications: when something fails in workflow we have to see what exactly has failed (unit tests or nimbus compile)  
- what benefits does current approach brings to outweigh split workflows?

Reply:
I did this mostly to avoid compiling nim-libp2p twice, and thought that we wouldn't want to try and compile it against nimbus unless tests passed, thus why it's where it is